artiq/applets/plot_xy.py

```diff
@@ -24,32 +31,57 @@ def data_changed(self, data, mods, title):
         fit = data.get(self.args.fit, (False, None))[1]
 
         if not len(y) or len(y) != len(x):
-            return
+            self.mismatch['X values'] = True
+            self.timer.start(1000)
+        else:
+            self.mismatch['X values'] = False
+            if not sum(self.mismatch.values()):
+                self.timer.stop()
         if error is not None and hasattr(error, "__len__"):
             if not len(error):
                 error = None
             elif len(error) != len(y):
-                return
+                self.mismatch['Error bars'] = True
+                self.timer.start(1000)
+            else:
+                self.mismatch['Error bars'] = False
+                if not sum(self.mismatch.values()):
+                    self.timer.stop()
         if fit is not None:
             if not len(fit):
                 fit = None
             elif len(fit) != len(y):
-                return
+                self.mismatch['Fit values'] = True
+                self.timer.start(1000)
+            else:
+                self.mismatch['Fit values'] = False
+                if not sum(self.mismatch.values()):
+                    self.timer.stop()
 
         self.clear()
+        if self.mismatch['X values']:
+            return
         self.plot(x, y, pen=None, symbol="x")
         self.setTitle(title)
-        if error is not None:
+        if error is not None and not self.mismatch['Error bars']:
```

Comment:
Why?

Reply:
The same as the checking ahead of `self.plot`, the applet window will crash if the length of error bars is mismatched with the length of Y values. And also the check of fit values.